workshops/read_the_internet/README.md

```diff
@@ -0,0 +1,175 @@
+        <td>Column data</td>
+        <td>Column 2 data</td>
+    </tr>
+    <tr>
+        <td>Canada</td>
+        <td>100,000</td>
+        <td>Column data</td>
+        <td>Column 2 data</td>
+    </tr>
+</table>
+```
+
+A HTML table can be seen as a 2 dimensional array or list. The table contains rows (`tr` tags). Each row contains columns (`td` tags). Within the `td` tags is the data or text we can webscrape.
+
+### Python time!
+
+We're going to use BeautifulSoup to find an HTML table element. We want to find a specific table by ID `sortable_table_world` as seen in the picture. The code below finds the first table that matches that ID by using the `find` function and specifying the id as an attribute. The `attrs` is an optional argument and is a dictionary that can be set to any id, class, or HTML identifier.
+
+![nCOVID table open with inspect element](https://cloud-p5pdve0kf.vercel.app/1inspect_element.png)
+
+```python
+# get world data table
+table = soup.find('table', attrs={'id':'sortable_table_world'})
+```
+After this code runs we now have a new beautiful soup object called `table` that contains the table and child elements within that table.
+
+With `table` we will use the `findAll` function to get all the `tr` tags from the table. `tableRows` is a Python list with a beautiful soup object for each `tr` tag found in the table.
+```python
+# get table rows
+tableRows = table.findAll('tr')
+```
+Now that we have all the rows in a Python list, we can iterate through the list and try to find the United States or you can input any Country you want.
+
+Before getting to the looping part though we will need to take a look at some of our outputs. This may happen to you when webscraping, the text has weird formatting. In these next steps I'm going to show the text from the website with the weird formatting and how we can use Python to remove that.
+
+I will just use the second row and first column for this example
+```python
+tableCols = tableRows[1].findAll('td')
+tableCol = tableCols[0]
+tableCol.text
+```
+This returns `'\r\n          TOTAL\r\n        '` with weird spacing and newline and indents. With Python we can use the `strip` function on the string and we will end with this code `tableCol.text.strip()` and this text `'TOTAL'`. Now that we got that handled lets get to reading the data!
+
+First we have a have a for each loop to loop through every row in the table. You make notice the `[1:]`, this was added to make the loop just a little more effcient because the first row at index `[0]` is just the table header with no data. Next we use an if statment to check if the current row's country is the United States (This can be replaced by any country). **Do Note** that we use the `.find()` function because we are only checking the first column and `.text[2:]` because the website has a weird astricks we need to remove. Once we find the country of our choice, we use `findAll()` to get all the columns in the row and print out some data.
```

Comment:
This is a very long paragraph with multiple explanations and instructions. It's very hard to follow. should be broken up